[
    {
        "description": "1. The naked_twins method applies the 'naked twins' strategy",
        "advice": "",
        "traceback": "Traceback (most recent call last):\n  File \"/home/grader/sudoku_solver/__utils.py\", line 135, in test\n    test_fn(*args, **kwargs)\n  File \"/home/grader/sudoku_solver/__utils.py\", line 73, in test\n    test_fn(*args, **kwargs)\n  File \"/home/grader/sudoku_solver/__utils.py\", line 354, in func_wrapper\n    return func(self)\n  File \"run3.py\", line 125, in test_naked_twins\n    prev_values = values.copy()\nAttributeError: 'NoneType' object has no attribute 'copy'\n",
        "result": "E",
        "message": "",
        "rubric_item_id": "10687"
    },
    {
        "description": "2. The solve method produces a valid solution",
        "advice": "Your code completed 0 test cases. The last test case executed was:\n9.1....8.8.5.7..4.2.4....6...7......5..............83.3..6......9................\n",
        "traceback": "Traceback (most recent call last):\n  File \"/home/grader/sudoku_solver/__utils.py\", line 135, in test\n    test_fn(*args, **kwargs)\n  File \"/home/grader/sudoku_solver/__utils.py\", line 73, in test\n    test_fn(*args, **kwargs)\n  File \"/home/grader/sudoku_solver/__utils.py\", line 354, in func_wrapper\n    return func(self)\n  File \"run3.py\", line 181, in test_solve\n    self.fail(msg=\"Your solve() function failed to find the solution.\", advice=advice)\n  File \"/home/grader/sudoku_solver/__utils.py\", line 278, in fail\n    super().fail(msg)\nAssertionError: Your solve() function failed to find the solution.\n",
        "result": "F",
        "message": "Your solve() function failed to find the solution.",
        "rubric_item_id": "10688"
    }
]
